### PR TITLE
Fix bug - Wrong approach for handling the bundles providing automatio…

### PR DESCRIPTION
**Pull request by adimova**
…n resources - host and fragments #1046

Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

**Comment by adimova** [comment]
#1094

File: bundles/automation/org.eclipse.smarthome.automation.providers/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -12,6 +12,8 @@ Import-Package:
  org.eclipse.smarthome.core.common.registry,
  org.eclipse.smarthome.core.i18n,
  org.osgi.framework,
+ org.osgi.framework.namespace;version="1.1.0",
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/META-INF/MANIFEST.MF]: could you remove the version constraints?

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     ],
     "configuration": {},
     "description": "Sample Rule for items definition.",
-    "on": [  
+    "triggers": [
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json]: Is this file not used by any test or how comes that no test was failing so far...?

**Comment by kaikreuzer** [comment]
Thanks @adimova. This goes very deep in the OSGi internals, I would not claim that I can follow every detail, but in general the code lgtm.
Nonetheless, I would feel better if there were some tests coming with this PR. Any chance to add one?

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json (hunk 2 above)
Comment (adimova) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json]: This file is used in Eclipse Smart Home Automation Integration Test and the test was failing before fix file

**Comment by adimova** [comment]
I made some manual testing to be sure that the code behave correct

**Comment by adimova** [comment]
the versions are removed in manifest ...e.smarthome.automation.providers/META-INF/MANIFEST.MF

**Comment by kaikreuzer** [comment]
> I made some manual testing to be sure that the code behave correct

This does not ensure that it will still be working if some changes are done to the code in future. I would prefer having this covered in an automated test.

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json (hunk 2 above)
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json]: No, this is why I am asking. So far no test is failing: https://hudson.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/lastCompletedBuild/testReport/

**Comment by danchom** [comment]
The file ItemRuleDefinition.json is part of org.eclipse.smarthome.automation.module.core.test, not from  automation integration test (org.eclipse.smarthome.automation.integration.test) and as I saw it is not used at all.

**Comment by kaikreuzer** [comment]
> as I saw it is not used at all.

Then we might better remove it...

**Comment by adimova** [comment]
PR is updated with new test method 'asserting that the host-fragment support works correctly' in AutomationIntegrationJsonTest and ...est/ESH-INF/automation/Xrules/ItemRuleDefinition.json is daleted

**Comment by kaikreuzer** [comment]
Thanks!

**Comment by kaikreuzer** [comment]
@adimova I seems that this PR broke the build: https://hudson.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/848/console
Could you please have a look?

**Comment by adimova** [comment]
frameworkWiring.refresh Bundles(null) is asynchronous operation and may be delayed from time to time, may be should waith more time

**Comment by adimova** [comment]
I fixed the Eclipse SmartHome Automation Integration Test - 'assert that a rule from json file is executed correctly' to wait enough

**Pull request merged**